fiftyone/core/dataset.py

```diff
@@ -262,8 +262,11 @@ def __init__(
         overwrite=False,
         _create=True,
         _virtual=False,
+        _doc=None,
```

Comment:
it seemed redundant to query the database to get the name to just then query the same document using the name, so I thought it would be nice to be able to init the dataset with the dataset doc just read from the db